src/Shared/TaskToApm.cs

```diff
@@ -42,7 +42,7 @@ public static void End(IAsyncResult asyncResult)
             return;
         }
 
-        throw new ArgumentNullException(nameof(asyncResult));
+        ArgumentNullException.ThrowIfNull(asyncResult, nameof(asyncResult));
```

Comment:
```suggestion
        ArgumentNullException.ThrowIfNull(asyncResult);
```

Reply:
Doesn't work here. Return value, and tests is more complex than just null

Reply:
? ThrowIfNull uses CallerArgumentExpressionAttribute to automatically do `nameof(x)` for you.